pinecone/client.go

```diff
@@ -1199,6 +1201,124 @@ func (c *Client) DeleteCollection(ctx context.Context, collectionName string) er
 	return nil
 }
 
+// EmbedRequest holds the parameters for generating embeddings for a list of input strings.
+//
+// Fields:
+//   - Model: (Required) The model to use for generating embeddings.
+//   - TextInputs: (Required) A list of strings to generate embeddings for.
+//   - Parameters: (Optional) EmbedParameters object that contains additional parameters to use when generating embeddings.
+type EmbedRequest struct {
+	Model      string
+	TextInputs []string
+	Parameters EmbedParameters
+}
+
+// EmbedParameters contains model-specific parameters that can be used for generating embeddings.
+//
+// Fields:
+//   - InputType: (Optional) A common property used to distinguish between different types of data. For example, "passage", or "query".
+//   - Truncate: (Optional) How to handle inputs longer than those supported by the model. if "NONE", when the input exceeds
+//     the maximum input token length an error will be returned.
```

Comment:
might want to put a comma after "length" just to make this sentence easier to grok